front/src/modules/trainschedule/components/ManageTrainSchedule/types.ts

```diff
@@ -0,0 +1,6 @@
+export type PowerRestrictionWarning = {
+  powerRestrictionCode?: string;
+  electrification?: string;
+  begin: number;
+  end: number;
+};
```

Comment:
You can make the type more precise
```suggestion
export type PowerRestrictionWarning = {
  begin: number;
  end: number;
} & (
  | {
      powerRestrictionCode: string;
      electrification: string;
    }
  | { powerRestrictionCode: typeof NO_POWER_RESTRICTION }
);
```

Reply:
Very good idea, thanks

Reply:
Actually, it adds a lot of complexity in PowerRestrictionSelector, so we decided to leave the type as it is.